applications/ShapeOptimizationApplication/python_scripts/communicator_factory.py

```diff
@@ -29,92 +29,98 @@ def __init__( self, optimization_settings ):
         self.__initializeListOfResponses()
 
     # --------------------------------------------------------------------------
-    def initializeCommunication( self ):
+    def initializeCommunication(self):
         self.__deleteAllRequests()
         self.__deleteAllReportedValues()
 
     # --------------------------------------------------------------------------
-    def requestValueOf( self, response_id ):
+    def requestValueOf(self, response_id):
         self.list_of_requests[response_id]["calculateValue"] = True
 
     # --------------------------------------------------------------------------
-    def requestGradientOf( self, response_id ):
+    def requestGradientOf(self, response_id):
         self.list_of_requests[response_id]["calculateGradient"] = True
 
     # --------------------------------------------------------------------------
-    def isRequestingValueOf( self, response_id ):
+    def isRequestingValueOf(self, response_id):
         if response_id not in self.list_of_requests.keys():
             return False
         else:
             return self.list_of_requests[response_id]["calculateValue"]
 
     # --------------------------------------------------------------------------
-    def isRequestingGradientOf( self, response_id ):
+    def isRequestingGradientOf(self, response_id):
         if response_id not in self.list_of_requests.keys():
             return False
         else:
             return self.list_of_requests[response_id]["calculateGradient"]
 
     # --------------------------------------------------------------------------
-    def reportValue( self, response_id, value ):
-        self.__storeValue( response_id, value)
-        if self.__isResponseWaitingForInitialValueAsReference( response_id ):
-            self.__setValueAsReference( response_id, value )
+    def reportValue(self, response_id, value):
+        self.__storeValue(response_id, value)
+        if self.__isResponseWaitingForInitialValueAsReference(response_id):
+            self.__setValueAsReference(response_id, value)
 
-        standardized_value = self.__translateValueToStandardForm( response_id, value )
-        self.__storeStandardizedValue( response_id, standardized_value )
+        standardized_value = self.__translateValueToStandardForm(response_id, value)
+        self.__storeStandardizedValue(response_id, standardized_value)
 
     # --------------------------------------------------------------------------
-    def reportGradient( self, response_id, gradient ):
-        standardized_gradient = self.__translateGradientToStandardForm( response_id, gradient )
-        self.__storeStandardizedGradient( response_id, standardized_gradient )
+    def reportGradient(self, response_id, gradient):
+        dimension = len(list(gradient.values())[0])
+        if dimension == 1:
+            gradient = {key: [value[0], 0.0, 0.0] for key, value in gradient.items()}
```

Comment:
Do we need this?

Reply:
Yes, since Kratos expects three values for both 2D and 3D cases. If, however, in the analyzer you are only provided with 2D gradients (e.g. as in SU2 if you run a 2D case), then you have to either fill the remaining 0 directly in the analyzer before you report or you do it manually in the communicator which was done here.

Reply:
I understand 2D and 3D but what would be the use case for 1D?